pkg/kubefed2/federate/util.go

```diff
@@ -106,24 +108,36 @@ func LookupAPIResource(config *rest.Config, key, targetVersion string) (*metav1.
 				lowerKey == strings.ToLower(resource.Kind) ||
 				lowerKey == fmt.Sprintf("%s.%s", resource.Name, gv.Group) {
 
-				targetResource = &resource
-				break
+				if targetResource != nil {
+					multiMatched = true
+				} else {
+					targetResource = resource.DeepCopy()
+					targetResource.Group = gv.Group
+					targetResource.Version = gv.Version
+				}
+
+				matchedResources = append(matchedResources, fmt.Sprintf("%s.%s", resource.Name, gv.Group))
```

Comment:
As per a previous comment, please use `pkg/apis/core/typeconfig:GroupQualifiedName` to compose the name to ensure that core resources (lacking a group) don't get an unnecessary suffix.

Reply:
Resolved. I just make a private function because currently it is only used in this file.